MapboxNavigation/RouteMapViewController.swift

```diff
@@ -248,19 +217,28 @@ class RouteMapViewController: UIViewController {
         self.mapView.setContentInset(self.contentInsets, animated: true, completionHandler: nil)
         self.mapView.setNeedsUpdateConstraints()
     }
+	
+    // MARK: - UIContentContainer
+	
+    override func preferredContentSizeDidChange(forChildContentContainer container: UIContentContainer) {
+        UIView.animate(withDuration: 0.3, animations: self.view.layoutIfNeeded)
+    }
 
+    // MARK: - RouteMapViewController
+	
     func notifyDidReroute(route: Route) {
-        updateETA()
+        self.updateETA()
         self.currentStepIndexMapped = 0
+        guard let routeController else { return }
```

Comment:
I think a guard statement like this is good, and preferable to having `routeController?` all throughout this method.

I was expecting it at the top though, or is there a reason we want to call `self.updateETA()` and mutate `currentStepIndexMapped` even when routeController is nil? 

As well as avoiding the work, I think it communicates more clearly "this method should only be called while actively routing"

Also, since this is a *re-route* do we expect routeController ever would actually be nil? If not, can you add an `assertionFailure` to the guard statement's else clause?

Reply:
good idea, it makes no sense to run call this method outside of a navigation, so routeController is required